### PR TITLE
Fix `create_overlay.yaml` workflow action

### PR DESCRIPTION
**Pull request by smiclea**
This fixes the incorrect overlay.yaml file name by removing an extra "$"
character.

Removes unnecessary dependency.

Uses correct variable names so that the output of the resource image is
not an empty string.

### CODE REVIEW

**Review by jnsgruk — APPROVED** [review]

**Comment by jnsgruk** [comment]
Hi @smiclea - thanks for all the changes! Can you please sign the CLA using the link from the bot above and let me know when you're covered? Thank you! /cc @maoo @TheJuanAndOnly99

File: .github/workflows/create_overlay.yaml [hunk 1]
```diff
@@ -44,40 +44,36 @@ jobs:
...
           echo "  legend-sdlc:" >> $overlay_file
           echo "    channel: ${{ steps.release-short.outputs.yyyy_mm }}/edge" >> $overlay_file
           echo "    resources:" >> $overlay_file
-          echo "      sdlc-image: ${{ github.event.inputs.sdlc_image }}" >> $overlay_file
+          echo "      sdlc-image: ${{ github.event.inputs.sdlc-image }}" >> $overlay_file
           echo "  legend-studio:" >> $overlay_file
           echo "    channel: ${{ steps.release-short.outputs.yyyy_mm }}/edge" >> $overlay_file
           echo "    resources:" >> $overlay_file
-          echo "      studio-image: ${{ github.event.inputs.studio_image }}" >> $overlay_file
+          echo "      studio-image: ${{ github.event.inputs.studio-image }}" >> $overlay_file
```
Comment (claudiubelu) [inline .github/workflows/create_overlay.yaml:70]: We still need to apply the overlay, see: https://github.com/finos/legend-juju-bundle/blob/main/.github/workflows/create_release.yaml#L45-L48

Comment (claudiubelu) [reply .github/workflows/create_overlay.yaml:70]: Actually, this action adds the overlay file, which can be later be used by people wanting to use specifically that release (we only have yyyy-mm tracks on charmhub, meaning that the second release in the month will override the first. The idea for the ``overlay.yaml`` file was to use that first release, if needed).

**Comment by smiclea** [comment]
> Hi @smiclea - thanks for all the changes! Can you please sign the CLA using the link from the bot above and let me know when you're covered? Thank you! /cc @maoo @TheJuanAndOnly99

I am now covered. Thank you!

**Pull request merged**